Sources/Attachment.php

```diff
@@ -1469,7 +1469,7 @@ public static function create(&$attachmentOptions)
 			);
 		}
 
-		if (!$this->is_image || empty(Config::$modSettings['attachmentThumbnails']) || (empty($attachmentOptions['width']) && empty($attachmentOptions['height']))) {
+		if (empty(Config::$modSettings['attachmentThumbnails']) || (empty($attachmentOptions['width']) && empty($attachmentOptions['height']))) {
```

Comment:
Why was this change made? If the attached file isn't an image, a thumbnail should never be made.

Reply:
`$this->image` doesn't exist.  I couldn't figure out why.  2.1 doesn't have this check.  Also we are in a static function, so `$this` can't exist.  We do have `$image` which is a object, but there is no `is_image` property.

Reply:
Changed it to check $image.  $image is set by the Image class which returns null if its not a image.  That should do what you are asking?

Reply:
I've commented on the new version below.